doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,653 @@
+~~~~~~~~~~~~~~~~~~~~~
+
+The details of the parameters will be added in this section. This section has
+the title "Parameters", followed by a line with a hyphen under each letter of
+the word "Parameters". A blank line is left before the section title, but not
+after, and not between the line with the word "Parameters" and the one with
+the hyphens.
+
+After the title, each parameter in the signature must be documented, including
+`*args` and `**kwargs`, but not `self`.
+
+The parameters are defined by their name, followed by a space, a colon, another
+space, and the type (or types). Note that the space between the name and the
+colon is important. Types are not defined for `*args` and `**kwargs`, but must
+be defined for all other parameters. After the parameter definition, it is 
+required to have a line with the parameter description, which is indented, and
+can have multiple lines. The description must start with a capital letter, and
+finish with a dot.
+
+Keyword arguments with a default value, the default will be listed in brackets
+at the end of the description (before the dot). The exact form of the
+description in this case would be "Description of the arg (default is X).". In
+some cases it may be useful to explain what the default argument means, which
+can be added after a comma "Description of the arg (default is -1, which means
+all cpus).".
+
+**Good:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, color='blue', **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Parameters
+            ----------
+            kind : str
+                Kind of matplotlib plot.
+            color : str
+                Color name or rgb code (default is 'blue').
+            **kwargs
+                These parameters will be passed to the matplotlib plotting
+                function.
+            """
+            pass
+
+**Bad:**
+
+.. code-block:: python
+
+    class Series:
+        def plot(self, kind, **kwargs):
+            """Generate a plot.
+
+            Render the data in the Series as a matplotlib plot of the
+            specified kind.
+
+            Note the blank line between the parameters title and the first
+            parameter. Also, not that after the name of the parameter `kind`
+            and before the colo, a space is missing.
+
+            Also, note that the parameter descriptions do not start with a
+            capital letter, and do not finish with a dot.
+
+            Finally, the `**kwargs` parameter is missing.
+
+            Parameters
+            ----------
+
+            kind: str
+                kind of matplotlib plot
+            """
+            pass
+
+Parameter types
+^^^^^^^^^^^^^^^
+
+When specifying the parameter types, Python built-in data types can be used
+directly:
+
+- int
+- float
+- str
+
+For complex types, define the subtypes:
+
+- list of [int]
```

Comment:
I think this is also something we are currently not doing? 
I am not sure if I find `list of [int]` better than `list of int`

(for the dict and tuple it can be more illustrative)

Reply:
Just trying to be consistent for list. I agree, in dict and tuple I think it adds value.

Probably not something it'd ever happen, but it could be clearer:
`list of [dict of {int: str}]`

than
`list of dict of {int: str}`

But happy with `list of int` too.